File: src/main/scala/com/zht/wc/WordCount.scala
package com.zht.wc

import org.apache.flink.api.scala.ExecutionEnvironment
import org.apache.flink.api.scala._           //这个是scala的所有隐式转换
/*
* 批处理wordcount
* */
object WordCount {
  def main(args: Array[String]): Unit = {

     //创建一个批处理执行环境
    val env =ExecutionEnvironment.getExecutionEnvironment


     //从文件读取数据
      val path = "D:\\soft\\idea_workplace\\FlinkWc\\src\\main\\resources\\hello.txt"

      val dataSet = env.readTextFile(path)
      val value: DataSet[String] = dataSet.flatMap(
        x=>(x.split(" "))
      )

    val result: DataSet[(String, Int)] = value.map(x => (x, 1))
    result.groupBy(0).sum(1).print()        //sum(1) 指的是下标为1



  }

}

File: src/main/scala/com/zht/transform/WordCount.java
package com.zht.transform;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.api.common.typeinfo.Types;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.connector.kafka.source.KafkaSource;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;
import org.apache.flink.util.Collector;

import java.util.Properties;

public class WordCount {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);
        DataStreamSource<String> lineds = env.readTextFile("datas/words.txt");

        Properties properties = new Properties();
        properties.setProperty("bootstrap.servers", "hadoop102:9092");
        properties.setProperty("group.id", "consumer-group");
        properties.setProperty("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        properties.setProperty("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        properties.setProperty("auto.offset.reset", "latest");

         env.addSource(new FlinkKafkaConsumer<String>("test",new SimpleStringSchema(),properties));

        SingleOutputStreamOperator<Tuple2<String, Long>> returns = lineds.flatMap((String line, Collector<Tuple2<String, Long>> out) -> {
            String[] split = line.split(" ");
            for (String word : split) {
                out.collect(Tuple2.of(word, 1L));
            }
        }).returns(Types.TUPLE(Types.STRING, Types.LONG));

        returns.keyBy(0).sum(1).print();

        env.execute();
    }
 }
